src/node_url.cc

```diff
@@ -853,9 +863,18 @@ static url_host_type ParseHost(url_host* host,
   // First, we have to percent decode
   PercentDecode(input, length, &decoded);
 
-  // Then we have to punycode toASCII
-  if (!ToASCII(&decoded, &decoded))
-    goto end;
+  // Match browser behavior for ASCII only domains
+  // and do not run them through ToASCII algorithm.
+  if (IsAllASCII(&decoded)) {
+    // Lowercase aschii domains
+    for (size_t n = 0; n < decoded.size(); n++) {
```

Comment:
For converting to lowercase, consider converting multiple bytes at a time instead of 1 by 1. [Here](https://github.com/client9/stringencoders/blob/196797afb2de11ae5381dad7246e25c7eeae2c1a/src/modp_ascii.c#L52) is probably a good example of how to do that.